docs/source/distributed.optim.rst

```diff
@@ -4,11 +4,8 @@
 Distributed Optimizers
 ======================
 
-.. autoclass:: torch.distributed.optim.DistributedOptimizer
-    :members:
+.. warning ::
+    Distributed optimizer is not currently supported when using CUDA tensors
```

Comment:
cc @wanchaol Do you recall why we had this limitation? I can't seem to think of a reason why CUDA tensors would not be supported?